cognite/fdm/scalars/timestamp.py

```diff
@@ -0,0 +1,132 @@
+from __future__ import annotations
+
+import logging
+import re
+from contextlib import suppress
+from datetime import datetime
+from typing import TYPE_CHECKING, Annotated, TypeVar
+
+logger = logging.getLogger(__name__)
+
+
+__all__ = [
+    "Timestamp",
+]
+
+
+TimestampParsable = TypeVar("TimestampParsable", str, datetime)
+
+
+class _Timestamp(str):
+    """
+    String in the format 'YYYY-MM-DDTHH:MM:SS[.millis][Z|time zone]' with optional milliseconds having precision
+    of 1-3 decimal digits and optional timezone with format ±HH:MM, ±HHMM, ±HH or Z, where Z represents UTC, Year
+    must be between 0001 and 9999.
```

Comment:
Why do we need to parse datetime-like strings ourselves? There have to be existing options for this right? 😬

Reply:
Parsing as part of validation, I want to only support the formats that can be written into FDM, not just any (valid or not) datetime format.

Reply:
Reworked this validation a bit, so that it doesn't change the value (it forced the isoformat before), now it just checks whether the value is in fact valid. With one exception: I'm allowing more then 3 digits of sub-second precision (i.e. decimal digits for second) but rounding it to 3 digits when needed. Seems reasonable to me... WDYT?

Reply:
Sounds very reasonable. I would technically floor it though, to not store it “into the future”. Nitpick > 9000

Reply:
Makes sense! Pushed.